week3/homework/src/index.js

```diff
@@ -1,3 +1,109 @@
 'use strict';
+const express = require('express');
+const uuid = require('uuid/v4');
 
-// TODO: Write the homework code in this file
+const { readFile: _readFile, writeFile: _writeFile } = require('fs');
+const { promisify } = require('util');
+
+const readFile = promisify(_readFile);
+const writeFile = promisify(_writeFile);
+
+const app = express();
+//build-in json middleware to automatically parse json
+app.use(express.json());
+const TODO_FILE = 'todos.json';
+
+function readTodos() {
+  return readFile(TODO_FILE, `utf8`).then(
+    JSON.parse,
+    () => ({})
+  );
+}
+
+function writeTodos(data) {
+  return writeFile(TODO_FILE, JSON.stringify(data));
+}
+
+//Create a todo
+app.post('/todos', async (req, res) => {
+  const newTodo = req.body;
+  const id = uuid();
+  newTodo.done = false;
+  const todos = await readTodos();
+  todos[id] = newTodo;
+  await writeTodos(todos);
+  res.json(todos);
+});
+
+//Read all todos
+app.get('/todos', async (req, res) => {
+  const todos = await readTodos();
+  res.json(todos);
+});
+
+//Read a todo by ID
+app.get('/todos/:id', async (req, res) => {
+  const id = req.params.id;
+  const todos = await readTodos();
+  if (id in todos) { res.json(todos[id]); }
+  else { res.json(`this ID isn't exist `); }
```

Comment:
When something is not found, you can also set the response status code to 404 to help clients easily figure out what went wrong.

Also, since we're expecting a JSON object back, you could wrap the errors into something like:

```json
{
  "error": "To-do not found"
}
```

Reply:
yes i think it's better too 
i'll see it & thank you for advice 💕
if there is anything too i'll be glad to learn more